config/ecs.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	ecsIpv4MaskMax = uint8(32)
+	ecsIpv6MaskMax = uint8(128)
+)
+
+// ECSv4Mask is the subnet mask to be added as EDNS0 option for IPv4
+type ECSv4Mask uint8
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface
+func (x *ECSv4Mask) UnmarshalText(text []byte) error {
+	res, err := unmarshalInternal(text, ecsIpv4MaskMax, "IPv4")
+	if err != nil {
+		return err
+	}
+
+	*x = ECSv4Mask(res)
+
+	return nil
+}
+
+// ECSv6Mask is the subnet mask to be added as EDNS0 option for IPv6
+type ECSv6Mask uint8
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface
+func (x *ECSv6Mask) UnmarshalText(text []byte) error {
+	res, err := unmarshalInternal(text, ecsIpv6MaskMax, "IPv6")
+	if err != nil {
+		return err
+	}
+
+	*x = ECSv6Mask(res)
+
+	return nil
+}
+
+// EcsConfig is the configuration of the ECS resolver
+type EcsConfig struct {
+	UseEcsAsClient bool      `yaml:"useEcsAsClient" default:"false"`
+	ForwardEcs     bool      `yaml:"forwardEcs" default:"false"`
```

Comment:
Maybe rename this to `forwardUpstream` or just `forward` since it's already under `ecs:`?

Reply:
Agreed `UseAsClient` & `Forward` would reduce naming redundancy.